### PR TITLE
Set course registration key to nil when duplicating

### LINKED ISSUE #2042: Duplication of course not working because of duplicated course registration code

### PR DESCRIPTION
**Pull request by fonglh**
Fixes #2042.

### CODE REVIEW

File: app/models/course.rb [hunk 1]
```diff
@@ -117,6 +117,7 @@ def initialize_duplicate(duplicator, other)
     self.start_at += duplicator.time_shift
     self.end_at += duplicator.time_shift
     self.title = duplicator.new_course_title
+    self.registration_key = nil
```
Comment (weiqingtoh) [inline app/models/course.rb]: Actually, should we generate a new registration key or just set it to `nil` by default?

Comment (fonglh) [inline app/models/course.rb]: I think it should be `nil`, the lecturer should explicitly setup registration code enrolment if he wants.

File: spec/services/course/duplication_service_spec.rb [hunk 2]
```diff
@@ -38,6 +38,11 @@
           expect(new_course.end_at).to be_within(1.second).of course.end_at + 1.day
         end
 
+        it 'sets new course registration key to nil' do
```
Comment (allenwq) [inline spec/services/course/duplication_service_spec.rb]: Test that a course with registration key can be duplicated?  It's not that helpful to just test what the code exactly does.

Comment (fonglh) [inline spec/services/course/duplication_service_spec.rb]: `new_course` is already the duplicated course.

Comment (fonglh) [inline spec/services/course/duplication_service_spec.rb]: Ok, it wasn't saved so no error gets thrown if it doesn't work. I combined this with the basic test for whether a course can be duplicated.

**Pull request merged**